homeassistant/components/open_meteo/weather.py

```diff
@@ -43,18 +55,53 @@ def __init__(
         *,
         entry: ConfigEntry,
         coordinator: DataUpdateCoordinator[OpenMeteoForecast],
+        hourly: bool,
     ) -> None:
         """Initialize Open-Meteo weather entity."""
         super().__init__(coordinator=coordinator)
         self._attr_unique_id = entry.entry_id
+        self._config = MappingProxyType(entry.data)
+        self._hourly = hourly
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        name = self._config.get(CONF_NAME)
+        name_appendix = ""
+        if self._hourly:
+            name_appendix = " hourly"
+
+        if name is not None:
+            return f"{name}{name_appendix}"
+
+        return f"{DEFAULT_NAME}{name_appendix}"
```

Comment:
This has `has_entity_name` so this should only be the appendix

Reply:
Might as well do it in `__init__` with `_attr_name`

Reply:
Do you mean like this?

`self._attr_name = "Hourly" if hourly else "Daily"`

Names then become 'Forecast Daily' and 'Forecast Hourly'

I have tried to follow the rules here: [has_entity_name True (Mandatory for new integrations)](https://developers.home-assistant.io/docs/core/entity/#entity-naming)